### ISSUE TITLE
YOLOConfiguration does not support number of channels!=3

### ISSUE THREAD

**Issue by kfirgedal** [PyTorchLightning/lightning-bolts#805]
Hi,

YOLOConfiguration implementation does not allow you to provide a number of channels in the .cfg file that is different than 3.

The line that is causing the problem is:
https://github.com/PyTorchLightning/lightning-bolts/blob/2415b49a2b405693cd499e09162c89f807abbdc4/pl_bolts/models/detection/yolo/yolo_config.py#L43

Thanks

**Comment by senarvi** [comment]
That's right, the number of channels in the input image is always 3 (RGB). @kfirgedal is there a way to specify the number of channels in the Darknet implementation?

**Comment by senarvi** [comment]
@kfirgedal indeed you're right. It should be trivial to add support for that. I can take a look at it soon, unless you want to create a pull request. I'm currently working on a big update that would bring features from YOLOv5, YOLOX, and Scaled-YOLOv4.
